### PR TITLE
test: improve test coverage of Google Maps adapter

### LINKED ISSUE #88: Unit test Google Maps Adapter
Adapters are the way in which we allow Terra Draw to work across different mapping libraries. An Adapter provides the code to map the library specific code to our more generalised Terra Draw interfaces so that all the Adapters can behave in the same way.

The unit test coverage for the Google Maps Adapter isn't where it should be, ideally we'd want to go for 80% +. It would be great to bring this up to speed! We have some guidance for new contributors [here](https://github.com/JamesLMilner/terra-draw/blob/main/guides/CONTRIBUTING.md) and also a guide on how to develop locally [here](https://github.com/JamesLMilner/terra-draw/blob/main/guides/DEVELOPMENT.md)

If you need inspiration on how to do this, you can see some example tested unit tests here:

ArcGIS Maps SDK Adapter:
https://github.com/JamesLMilner/terra-draw/blob/main/src/adapters/arcgis-maps-sdk.adapter.spec.ts

Leaflet.js Adapter:
https://github.com/JamesLMilner/terra-draw/blob/main/src/adapters/leaflet.adapter.ts

### PR DESCRIPTION
**Pull request by AndrewADev**
# Improve Test coverage of Google Maps Adapter

Add more tests for the Google Maps adapter. With these changes, coverage by statements and by lines are now just over 80% for the Google Maps adapter.

Closes #88 

## Details

Note that these changes add [JSDOM](https://github.com/jsdom/jsdom) to enable DOM emulation for the tests. This was the [simplest](https://github.com/JamesLMilner/terra-draw/issues/88#issuecomment-1745749984) way to get the tests up and running for this adapter. For now I've only enabled the JSDOM environment for the Google Maps adapter (via the `@jest-environment` [directive in the docblock](https://jestjs.io/docs/configuration#testenvironmentoptions-object) ), so the environment remains the same (still [`node`](https://github.com/JamesLMilner/terra-draw/blob/main/jest.config.ts#L14)) for the other modules.

In addition to JSDOM, I also considered [Happy DOM](https://github.com/capricorn86/happy-dom), which worked just fine on a functional level, but wasn't consistently faster during casual testing (no real performance testing done).<sup>1</sup> Nonetheless, it wouldn't be hard to add it later on - just install the `@happy-dom/jest-environment` package, and specify it as the `@jest-environment` at the top of the file) 

## Verification

The changes themselves are either tests, or minor refactorings to make the library code more testable, and all tests are passing.

Beyond that, I set up a Google Maps API key, added it to the local `.env` file, and briefly clicked through the Terra Draw Google Maps instance locally. I was able to add & remove shapes and anotations without errors:   
image

<sup>1</sup> - _Casual observation demonstrated similarly large variance with both JSDOM and Happy DOM, likely more tied to the number of background processes that happened to be running on my development machine than anything else._

### CODE REVIEW

File: src/adapters/google-maps.adapter.ts [hunk 1]
```diff
@@ -59,12 +59,12 @@ export class TerraDrawGoogleMapsAdapter extends TerraDrawBaseAdapter {
 
 		const ne = bounds.getNorthEast();
 		const sw = bounds.getSouthWest();
-		const latLngBounds = new google.maps.LatLngBounds(sw, ne);
+		const latLngBounds = new this._lib.LatLngBounds(sw, ne);
```
Comment (JamesLMilner) [inline src/adapters/google-maps.adapter.ts:62]: Thanks for noticing these - this was not intentional and should have been `this._lib` from the start!

File: src/adapters/google-maps.adapter.spec.ts [hunk 2]
```diff
@@ -64,4 +80,938 @@ describe("TerraDrawGoogleMapsAdapter", () => {
...
+			adapter.setCursor("pointer");
+			adapter.setCursor("pointer");
+			adapter.setCursor("pointer");
+
+			expect(map.getDiv).toHaveBeenCalledTimes(1);
+		});
+	});
+
+	describe("render", () => {
+		it("first invocation registers event listeners", () => {
```
Comment (AndrewADev) [inline src/adapters/google-maps.adapter.spec.ts:457]: I was actually slightly surprised that `render` seemed to take on an 'initialization' role here (registering event listeners the first time it is called). The ArcGIS adapter, for instance, does this via a separate method, [invoked in the constructor](https://github.com/JamesLMilner/terra-draw/blob/main/src/adapters/arcgis-maps-sdk.adapter.ts#L59). Would you be open to refactoring changes to move the 'init' code to a separate method, and invoking in the constructor ?

**Comment by AndrewADev** [comment]
@JamesLMilner I verified these changes in the dev instance locally, and have added details on my verification steps to description, so these changes should now be ready to review.

File: src/adapters/google-maps.adapter.spec.ts (hunk 2 above)
Comment (JamesLMilner) [reply src/adapters/google-maps.adapter.spec.ts:457]: Yes this is a fair observation and you're probably right that these would be better suited in the constructor. I think lets look into as a separate issue as not to block your hard work?

**Review by JamesLMilner — APPROVED** [review]
Amazing work, this looks good to go! I would just make one small request which is if you would kindly perhaps condense this down to one commit? i.e. `test: improve test coverage of Google Maps adapter` - this just helps make the commit history a bit cleaner as it drives the change log.

File: src/adapters/google-maps.adapter.spec.ts (hunk 2 above)
Comment (AndrewADev) [reply src/adapters/google-maps.adapter.spec.ts:457]: yes, was thinking of a separate set of changes

**Comment by AndrewADev** [comment]
> Amazing work, this looks good to go! I would just make one small request which is if you would kindly perhaps condense this down to one commit? i.e. `test: improve test coverage of Google Maps adapter` - this just helps make the commit history a bit cleaner as it drives the change log.

done

**Pull request merged**